py/dml/messages.py

```diff
@@ -1891,6 +1891,12 @@ def log(self):
             "conflicting declaration, which specifies the type: "
             + self.other_type.describe())
 
+
+class EVECT(DMLError):
+    """`vect` types are only permitted if the [`c_vect` provisional
+    feature](provisional-auto.html#c_vect) is enabled."""
+    fmt = "declaration of vect type without c_vect provisional"
```

Comment:
I did consider `legacy_vect`, but it's potentially misleading since `vect` is not considered a "legacy feature" in the Simics sense. I picked `c_vect` because it's a thin wrapping around some C macros, and because `c` is short and distinguishes it from RAII.

Maybe sufficient to rename `EVECT` to `ECVECT` or `EC_VECT`?
(I chose EVECT because it's not yet ambiguous, and it's cheap to rename error classes the day an ambiguity is introduced)

Reply:
`ECVECT` is better than `EVECT`.

I'm not... _super_-fond of `c_vect`, but I think I could accept it. In truth, we want a solid connection to vect.h, and there are various ways one can put that: Simics vectors; `VECT` vectors; "macro vectors"; and, of course, the one I most commonly use, "vect.h". It feels to me that "C vectors" are the weakest of the lot due to the lack of specificity.

Reply:
What about `simics_util_vect`?

Reply:
Now renamed to `ECVECT`

Reply:
Renamed to your proposed simics_util_vect after a user poll elsewhere

Reply:
Oh, no wait, two votes.